### ISSUE TITLE
Oculus Quest "Failed to start" (remote server)

### ISSUE THREAD

**Issue by jeffkingdev** [baldurk/renderdoc#1399]
## Description

Can't connect renderdoc to my oculus quest.  Says "Remote server not running or failed to start". 

Please note, I ran "adb shell pm list packages org.renderdoc.renderdoccmd" and I see:
package:org.renderdoc.renderdoccmd.arm32
package:org.renderdoc.renderdoccmd.arm64

So, it appears it installed.  It's just not running for some reason.  Any ideas?  Do I need a special version of Renderdoc for the Oculus Quest?

Here's the output
05-31 08:35:49.948  4902  4902 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
05-31 08:35:49.950  4902  4902 D AndroidRuntime: CheckJNI is OFF
05-31 08:35:50.034  4902  4902 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
05-31 08:35:50.050  4902  4902 I Radio-JNI: register_android_hardware_Radio DONE
05-31 08:35:50.059  4902  4902 D AndroidRuntime: Calling main entry com.android.commands.pm.Pm
05-31 08:35:50.069  4902  4902 I art     : System.exit called, status: 0
05-31 08:35:50.069  4902  4902 I AndroidRuntime: VM exiting with result code 0.
05-31 08:35:50.072  4902  4910 W MessageQueue: Handler (android.os.Handler) {4941d43} sending message to a Handler on a dead thread
05-31 08:35:50.072  4902  4910 W MessageQueue: java.lang.IllegalStateException: Handler (android.os.Handler) {4941d43} sending message to a Handler on a dead thread
05-31 08:35:50.072  4902  4910 W MessageQueue:  at android.os.MessageQueue.enqueueMessage(MessageQueue.java:543)
05-31 08:35:50.072  4902  4910 W MessageQueue:  at android.os.Handler.enqueueMessage(Handler.java:643)
05-31 08:35:50.072  4902  4910 W MessageQueue:  at android.os.Handler.sendMessageAtTime(Handler.java:612)
05-31 08:35:50.072  4902  4910 W MessageQueue:  at android.os.Handler.sendMessageDelayed(Handler.java:582)
05-31 08:35:50.072  4902  4910 W MessageQueue:  at android.os.Handler.post(Handler.java:338)
05-31 08:35:50.072  4902  4910 W MessageQueue:  at android.os.ResultReceiver$MyResultReceiver.send(ResultReceiver.java:57)
05-31 08:35:50.072  4902  4910 W MessageQueue:  at com.android.internal.os.IResultReceiver$Stub.onTransact(IResultReceiver.java:58)
05-31 08:35:50.072  4902  4910 W MessageQueue:  at android.os.Binder.execTransact(Binder.java:565)
05-31 08:35:50.394   705   705 D wifi_sig_helper: Wifi rssi: -32
05-31 08:35:50.537  4916  4916 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
05-31 08:35:50.540  4916  4916 D AndroidRuntime: CheckJNI is OFF
05-31 08:35:50.609  4916  4916 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
05-31 08:35:50.627  4916  4916 I Radio-JNI: register_android_hardware_Radio DONE
05-31 08:35:50.636  4916  4916 D AndroidRuntime: Calling main entry com.android.commands.pm.Pm
05-31 08:35:50.706  1152  1468 I ProcessStatsService: Added stats: 2019-05-31-04-50-57, over +3h3m46s842ms
05-31 08:35:50.710  1152  1468 I ProcessStatsService: Added stats: 2019-05-31-01-45-10, over +3h5m46s646ms
05-31 08:35:50.715  1152  1468 I ProcessStatsService: Added stats: 2019-05-30-22-36-10, over +3h9m0s506ms
05-31 08:35:50.720  1152  1468 I ProcessStatsService: Added stats: 2019-05-30-19-28-41, over +3h7m29s400ms
05-31 08:35:50.724  1152  1468 I ProcessStatsService: Added stats: 2019-05-30-16-20-49, over +3h7m51s460ms
05-31 08:35:50.728  1152  1468 I ProcessStatsService: Added stats: 2019-05-30-13-18-19, over +3h2m30s646ms
05-31 08:35:50.732  1152  1468 I ProcessStatsService: Added stats: 2019-05-30-10-12-12, over +3h6m7s472ms
05-31 08:35:50.736  1152  1468 I ProcessStatsService: Added stats: 2019-05-30-07-05-05, over +3h7m6s504ms
05-31 08:35:50.761  1152  1468 I ProcessStatsService: Added stats: 2019-05-31-04-50-57, over +3h3m46s842ms
05-31 08:35:50.865  1152  1553 E BatteryStatsService: no controller energy info supplied
05-31 08:35:50.883  4916  4916 I art     : System.exit called, status: 0
05-31 08:35:50.883  4916  4916 I AndroidRuntime: VM exiting with result code 0.
05-31 08:35:51.247  4935  4935 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
05-31 08:35:51.250  4935  4935 D AndroidRuntime: CheckJNI is OFF
05-31 08:35:51.315  4935  4935 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
05-31 08:35:51.331  4935  4935 I Radio-JNI: register_android_hardware_Radio DONE
05-31 08:35:51.339  4935  4935 D AndroidRuntime: Calling main entry com.android.commands.am.Am
05-31 08:35:51.341  1152  1162 I ActivityManager: Force stopping org.renderdoc.renderdoccmd.arm32 appid=10083 user=0: from pid 4935
05-31 08:35:51.342  1152  1162 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:51.343  4935  4935 D AndroidRuntime: Shutting down VM
05-31 08:35:51.351   693  1241 D SettingsService: Setting value retrieved from db [crash_reports_enabled : 1]
05-31 08:35:51.407   705   705 D wifi_sig_helper: Wifi rssi: -31
05-31 08:35:51.700  4945  4945 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
05-31 08:35:51.703  4945  4945 D AndroidRuntime: CheckJNI is OFF
05-31 08:35:51.770  4945  4945 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
05-31 08:35:51.788  4945  4945 I Radio-JNI: register_android_hardware_Radio DONE
05-31 08:35:51.797  4945  4945 D AndroidRuntime: Calling main entry com.android.commands.am.Am
05-31 08:35:51.799  1152  1163 I ActivityManager: Force stopping org.renderdoc.renderdoccmd.arm64 appid=10084 user=0: from pid 4945
05-31 08:35:51.799  1152  1163 I ActivityManager: Killing 4866:org.renderdoc.renderdoccmd.arm64/u0a84 (adj 0): stop org.renderdoc.renderdoccmd.arm64
05-31 08:35:51.799  1152  1163 D ActivityManager: cleanUpApplicationRecord -- 4866
05-31 08:35:51.800  1152  1163 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:51.806  1152  1191 W InputDispatcher: channel '4da9276 org.renderdoc.renderdoccmd.arm64/org.renderdoc.renderdoccmd.arm64.Loader (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
05-31 08:35:51.807  1152  1191 E InputDispatcher: channel '4da9276 org.renderdoc.renderdoccmd.arm64/org.renderdoc.renderdoccmd.arm64.Loader (server)' ~ Channel is unrecoverably broken and will be disposed!
05-31 08:35:51.807  1152  1494 I WindowManager: WIN DEATH: Window{4da9276 u0 org.renderdoc.renderdoccmd.arm64/org.renderdoc.renderdoccmd.arm64.Loader}
05-31 08:35:51.807  1152  1494 W InputDispatcher: Attempted to unregister already unregistered input channel '4da9276 org.renderdoc.renderdoccmd.arm64/org.renderdoc.renderdoccmd.arm64.Loader (server)'
05-31 08:35:51.810   693   693 D SettingsService: Setting value retrieved from db [crash_reports_enabled : 1]
05-31 08:35:51.815  1152  1163 I ActivityManager: Start proc 4955:org.renderdoc.renderdoccmd.arm64/u0a84 for activity org.renderdoc.renderdoccmd.arm64/.Loader
05-31 08:35:51.815  1152  1163 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:51.818  1152  1163 I ActivityManager:   Force finishing activity ActivityRecord{c78a787 u0 org.renderdoc.renderdoccmd.arm64/.Loader t98}
05-31 08:35:51.820  4955  4955 I art     : Late-enabling -Xcheck:jni
05-31 08:35:51.822   693  1241 D SettingsService: Setting value retrieved from db [crash_reports_enabled : 1]
05-31 08:35:51.822  3930  3930 I panelapp: onTrim 80
05-31 08:35:51.823  1152  1163 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:51.826  1152  1163 I ActivityManager:   Force finishing activity ActivityRecord{c78a787 u0 org.renderdoc.renderdoccmd.arm64/.Loader t98 f}
05-31 08:35:51.826  1152  1163 W ActivityManager: Duplicate finish request for ActivityRecord{c78a787 u0 org.renderdoc.renderdoccmd.arm64/.Loader t98 f}
05-31 08:35:51.829  4945  4945 D AndroidRuntime: Shutting down VM
05-31 08:35:51.833   693   693 D SettingsService: Setting value retrieved from db [crash_reports_enabled : 1]
05-31 08:35:51.891  1837  1837 D [OAR] MainActivity: com.oculus.vrshell.MainActivity@b92aa4b surfaceCreated()
05-31 08:35:51.891  1837  1837 I [OAR] ShellMain:     looperMessageQueue->SendPrintf("surfaceCreated ")
05-31 08:35:51.891  1837  1906 I [OAR] ShellApp: 0xcaf7f2d0 msg: surfaceCreated
05-31 08:35:51.891  1837  1837 D [OAR] MainActivity: com.oculus.vrshell.MainActivity@b92aa4b surfaceChanged() format: 4 width: 2880 height: 1600
05-31 08:35:51.892  1837  1837 I [OAR] ShellMain:     ANativeWindow_release(newNativeWindow)
05-31 08:35:51.898  2071  2071 D VrRuntimeService: onFocusedWindowChanged: topWindowToken=android.os.BinderProxy@92870fc, topWindowUid=10024
05-31 08:35:51.898  2071  2071 W ClientMgr: ClientMgr::SetActiveCompositorClient: No active client
05-31 08:35:51.898  2071  2071 W VrRuntimeServer: VrRuntimeServer_SetClockLevels: Invalid Warp when trying to set -1 -1
05-31 08:35:51.898  1837  2187 I OpenGLRenderer: Initialized EGL, version 1.4
05-31 08:35:51.899  1837  2187 D OpenGLRenderer: Swap behavior 1
05-31 08:35:52.414  4972  4972 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
05-31 08:35:52.417   705   705 D wifi_sig_helper: Wifi rssi: -33
05-31 08:35:52.418  4972  4972 D AndroidRuntime: CheckJNI is OFF
05-31 08:35:52.497  4972  4972 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
05-31 08:35:52.515  4972  4972 I Radio-JNI: register_android_hardware_Radio DONE
05-31 08:35:52.524  4972  4972 D AndroidRuntime: Calling main entry com.android.commands.am.Am
05-31 08:35:52.529  1152  1466 D NonVrActivityIntentWrapper: Is Intent { flg=0x10000000 cmp=org.renderdoc.renderdoccmd.arm64/.Loader (has extras) }, a valid launch intent: false
05-31 08:35:52.529  1152  1466 I ActivityManager: START u0 {flg=0x10000000 cmp=org.renderdoc.renderdoccmd.arm64/.Loader (has extras)} from uid 2000 on display 0
05-31 08:35:52.532  1152  1466 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.534  1152  1466 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.550  1152  1466 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.555  4972  4972 D AndroidRuntime: Shutting down VM
05-31 08:35:52.556  1837  1837 D [OAR] MainActivity: com.oculus.vrshell.MainActivity@b92aa4b surfaceDestroyed()
05-31 08:35:52.556  1837  1837 I [OAR] ShellMain:     looperMessageQueue->SendPrintf("surfaceDestroyed ")
05-31 08:35:52.557  1837  1906 I [OAR] ShellApp: 0xcaf7f2d0 msg: surfaceDestroyed
05-31 08:35:52.557  1837  1837 I [OAR] ShellMain:     ANativeWindow_release(nativeWindow)
05-31 08:35:52.564   693  1241 D SettingsService: Setting value retrieved from db [crash_reports_enabled : 1]
05-31 08:35:52.574  4955  4955 I renderdoc: @5cf12db800000001@ RDOC 004955: [08:35:52]             core.cpp( 351) - Log     - RenderDoc v1.4 Android 64-bit Release (214d85228538e71cc63a0d7fa11dd75b1d56cc81) loaded in replay application
05-31 08:35:52.574  4955  4955 I renderdoccmd: Creating lock m_DrawLock
05-31 08:35:52.574  4955  4955 I renderdoccmd: Creating lock m_CmdLock
05-31 08:35:52.614  4955  4983 I renderdoccmd: android_main android_state->window: 0x0
05-31 08:35:52.615  1152  1523 I ActivityManager: START u0 {act=android.content.pm.action.REQUEST_PERMISSIONS pkg=com.android.packageinstaller cmp=com.android.packageinstaller/.permission.ui.GrantPermissionsActivity (has extras)} from uid 10084 on display 0
05-31 08:35:52.618  1152  1523 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.623  4955  4983 I renderdoccmd: handle_cmd(10)
05-31 08:35:52.626  4955  4983 I renderdoccmd: handle_cmd(11)
05-31 08:35:52.633  4955  4983 I renderdoccmd: handle_cmd(0)
05-31 08:35:52.634  4955  4983 I renderdoccmd: handle_cmd(13)
05-31 08:35:52.634  1152  1521 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.636  1152  1521 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.640  1152  1521 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.643  1152  1521 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.655  4955  4983 I renderdoccmd: handle_cmd(12)
05-31 08:35:52.655  4955  4983 I renderdoccmd: handle_cmd(14)
05-31 08:35:52.665  2522  2522 I VrAppLoader: VrApi Loader Version: 1.1.15.0-0-0 Jun 11 2018 11:32:55
05-31 08:35:52.670  2522  2522 V VrDriver: Loading 64-bit VrDriver.
05-31 08:35:52.671  2522  2522 I DriverLoader: *Adding Layer: PowerLevelSwitcher
05-31 08:35:52.671  2522  2522 I DriverLoader: *Adding Layer: UtilPoller
05-31 08:35:52.671  2522  2522 I DriverLoader: About to call StartLayer in vrapilayers.UtilPoller
05-31 08:35:52.671  2522  2522 D UtilPollerJava: UtilPoller StartLayer: Loading utilpoller.so
05-31 08:35:52.672  2522  2522 I VrApiLayers: Initializing VrApiLayer
05-31 08:35:52.672  2522  2522 I DriverLoader: VrapiLayer UtilPoller started
05-31 08:35:52.672  2522  2522 I DriverLoader: About to call StartLayer in vrapilayers.PowerLevelSwitcher
05-31 08:35:52.673  2522  2522 D PowerLevelSwitcherJava: PowerLevelSwitcher StartLayer: Loading powerlevelswitcher.so
05-31 08:35:52.673  2522  2522 I VrApiLayers: Initializing VrApiLayer
05-31 08:35:52.673  2522  2522 I DriverLoader: VrapiLayer PowerLevelSwitcher started
05-31 08:35:52.673  2522  2522 I VrApi   : ----------- vrapi_Initialize [start] ----------
05-31 08:35:52.674  2522  2522 I Adreno  : QUALCOMM build                   : b1f8328, I8a36212832
05-31 08:35:52.674  2522  2522 I Adreno  : Build Date                       : 03/05/19
05-31 08:35:52.674  2522  2522 I Adreno  : OpenGL ES Shader Compiler Version: EV031.26.02.00
05-31 08:35:52.674  2522  2522 I Adreno  : Local Branch                     :
05-31 08:35:52.674  2522  2522 I Adreno  : Remote Branch                    : refs/tags/AU_LINUX_ANDROID_LA.UM.5.7.R1.07.01.01.253.109
05-31 08:35:52.674  2522  2522 I Adreno  : Remote Branch                    : NONE
05-31 08:35:52.674  2522  2522 I Adreno  : Reconstruct Branch               : NOTHING
05-31 08:35:52.674  2522  2522 I Adreno  : Build Config                     : S L 3.8.9 AArch64
05-31 08:35:52.681  2522  2522 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.682  2522  2522 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.683  2522  2522 I Adreno  : PFP: 0x005ff112, ME: 0x005ff066
05-31 08:35:52.683  2522  2522 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.685  2522  2522 I DeviceProperties_Android: ovrDeviceProperties::Initialize
05-31 08:35:52.686  2522  2522 I DeviceProperties_Android: Headset Model: 'monterey:SDC:SDINDDH4-64G    '
05-31 08:35:52.686  2522  2522 I DeviceProperties_Android: Device Model: 'monterey'
05-31 08:35:52.687  2522  2522 I DeviceProperties_Android: Device Region: 'US'
05-31 08:35:52.687  2522  2522 I DeviceProperties_Android: GPU Type: 'ADRENO_540'
05-31 08:35:52.688  2522  2522 I DeviceProperties_Android: Num Refresh Rates Supported 1
05-31 08:35:52.688  2522  2522 I DeviceProperties_Android: Refresh Rate (0): 72.000000Hz
05-31 08:35:52.688  2522  2522 I OSSDK   : CalibrationManager: Using standalone implementation
05-31 08:35:52.691  2522  2522 I DeviceProperties_Android: DisplayParms.ScanoutDirection = 3
05-31 08:35:52.691  2522  2522 I DeviceProperties_Android: DisplayParms.ShutterType = 0
05-31 08:35:52.691  2522  2522 I DeviceProperties_Android: DisplayParms.timing = 0.500000 1.000000
05-31 08:35:52.691  2522  2522 I DeviceProperties_Android: DisplayParms.prediction = (1.000000 1.500000) (1.500000 2.000000)
05-31 08:35:52.691  2522  2522 I DeviceProperties_Android: getDisplayDimensions returned: 2880, 1600
05-31 08:35:52.691  2522  2522 I Breakpad: Breakpad Init
05-31 08:35:52.691  2522  2522 I Breakpad: Breakpad for Native crashes disabled
05-31 08:35:52.691  2522  2522 D SystemActivitiesReceiver: Registering Oculus System Activity receiver
05-31 08:35:52.697  2522  2522 I SystemUtils_Internal: ShellSystemUtilsAnytimeUICapability: 2
05-31 08:35:52.697  2522  2522 I VrRuntimeClient: VrRuntimeClient: Init
05-31 08:35:52.697  2522  2522 I VrRuntimeClient: RuntimeService Enabled
05-31 08:35:52.697  2522  2522 I RuntimeHelpers: ShouldEnableOOPComposition: oopc_enabled_prop = -1
05-31 08:35:52.697  2522  2522 I RuntimeHelpers: ShouldEnableOOPComposition: enabled 1
05-31 08:35:52.697  2522  2522 I RuntimeHelpers: IsOopCompositionEnabled: enabled
05-31 08:35:52.697  2522  2522 I RuntimeHelpers: OOP_INPUT: oopi_enabled_prop = 1
05-31 08:35:52.703  2522  2522 I VrRuntimeClient: InsideFlailLockoutGKorSysProp true
05-31 08:35:52.703  2522  2522 I VrRuntimeClient: Flail Lockout Hand: 'combined'
05-31 08:35:52.703  2522  2522 I VrRuntimeClient: Flail Lockout Slow Test: '50'  Change '0.05'
05-31 08:35:52.703  2522  2522 I VrRuntimeClient: Flail Lockout Fast Test: '100'  Change '0.1'
05-31 08:35:52.704  2522  2522 D VrRuntimeClient: VrRuntimeClient: initialize. inContext: com.oculus.os.DialogContext@10d979c
05-31 08:35:52.705  2071  2071 D VrRuntimeService: onStartCommand
05-31 08:35:52.705  2522  2522 D VrRuntimeClient: connectToService
05-31 08:35:52.705  2522  2522 D VrRuntimeClient: VrRuntimeClient: creating new service
05-31 08:35:52.706  2071  2071 D VrRuntimeService: onStartCommand
05-31 08:35:52.707  2522  2522 D VrRuntimeClient: VrRuntimeClient: Waiting on connection...
05-31 08:35:52.707  2522  4984 D VrRuntimeClient: VrRuntimeClient: Service connected.
05-31 08:35:52.708  2071  3431 I ClientMgr: ClientMgr::CreateClient: 2522:com.android.packageinstaller
05-31 08:35:52.708  2071  3431 I SharedMemoryServer: SharedMemoryServer::MapMemory: Allocated shared memory fd 40 (VrRuntimeService_PrivateReadWrite_com.android.packageinstaller)
05-31 08:35:52.708  2071  3431 W Vulkan_Utils: Memory type 24 with properties 10 not found.
05-31 08:35:52.709  2071  3431 I BufferObject: ovrBufferObjectFactory::Initialize BufferSize=4096, Fd=44, AllocSize=9568, VkBuffer=0x7f73c22500, VkMemory=0x7f73caee00, Mapping=0x7f906fc000
05-31 08:35:52.709  2071  3431 I GuardianSystem: SetCurrentPackageName: com.android.packageinstaller
05-31 08:35:52.709  2071  2085 D VrRuntimeService: IVrRuntimeClient.Stub().getSharedMemoryFileDescriptor()
05-31 08:35:52.709  2071  2085 I VrRuntimeService: SharedMemoryDebug: nativeGetSharedMemoryPublicReadOnlyFd: 34
05-31 08:35:52.709  2071  2085 D VrRuntimeService: getSharedMemoryFileDescriptor fd (VrRuntimeService) 34
05-31 08:35:52.710  2071  2274 D VrRuntimeService: IVrRuntimeClient.Stub().getSharedMemoryPrivateReadWriteFileDescriptor()
05-31 08:35:52.710  2071  2274 I VrRuntimeService: SharedMemoryDebug: nativeGetSharedMemoryPrivateReadWriteFd: 40
05-31 08:35:52.710  2071  2274 D VrRuntimeService: getSharedMemoryPrivateReadWriteFileDescriptor fd (VrRuntimeService) 40
05-31 08:35:52.710  2071  2086 D VrRuntimeService: IVrRuntimeClient.Stub().getBufferObjectImportInfo()
05-31 08:35:52.710  2522  4984 I VrRuntimeClient: nativeRuntimeServiceConnected: publicReadOnlyFd=41, privateReadWriteFd=42, bufferObjectFd=43
05-31 08:35:52.711  2522  2522 D VrRuntimeClient: VrRuntimeClient: connectionIsReady: true
05-31 08:35:52.711  2522  2522 D VrRuntimeClient: VrRuntimeClient: context.bindService SUCCESS
05-31 08:35:52.711  2522  2522 I AudioCapture: AudioCapture::Init
05-31 08:35:52.711  2522  2522 I AudioCapture: Initializing audio capture using direct OS Callbacks
05-31 08:35:52.711  2522  2522 I OSSDK   : CalibrationManager: Using standalone implementation
05-31 08:35:52.714  2522  2522 I Distortion: Loaded distortion file from Calibration Manager
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::magic = 0x56347807
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::deviceModel = 0x101
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::headsetModel = 0x103
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeBlocksWide = 32
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeBlocksHigh = 32
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::lensSeparationMeters = 0.053428
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::horizontalOffsetMeters = 0.000000
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::displayWidthMeters = 0.118800
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::displayHeightMeters = 0.066000
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::displayWidthPixels = 2880
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::displayHeightPixels = 1600
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeTextureWidthPixels = 1216
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeTextureHeightPixels = 1344
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::leftEye = up 47.000000 / down 53.000000 / left 52.000000 / right 42.000000
05-31 08:35:52.714  2522  2522 I Distortion: ovrDistortionMeshHeader::rightEye = up 47.000000 / down 53.000000 / left 42.000000 / right 52.000000
05-31 08:35:52.714  2522  2522 I OSSDK   : CalibrationManager: Using standalone implementation
05-31 08:35:52.716  2522  2522 I Distortion: Loaded distortion file from Calibration Manager
05-31 08:35:52.716  2522  2522 I Distortion: ovrDistortionMeshHeader::magic = 0x56347807
05-31 08:35:52.716  2522  2522 I Distortion: ovrDistortionMeshHeader::deviceModel = 0x101
05-31 08:35:52.716  2522  2522 I Distortion: ovrDistortionMeshHeader::headsetModel = 0x103
05-31 08:35:52.716  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeBlocksWide = 32
05-31 08:35:52.716  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeBlocksHigh = 32
05-31 08:35:52.716  2522  2522 I Distortion: ovrDistortionMeshHeader::lensSeparationMeters = 0.053428
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::horizontalOffsetMeters = 0.000000
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::displayWidthMeters = 0.118800
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::displayHeightMeters = 0.066000
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::displayWidthPixels = 2880
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::displayHeightPixels = 1600
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeTextureWidthPixels = 1216
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::eyeTextureHeightPixels = 1344
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::leftEye = up 47.000000 / down 53.000000 / left 52.000000 / right 42.000000
05-31 08:35:52.717  2522  2522 I Distortion: ovrDistortionMeshHeader::rightEye = up 47.000000 / down 53.000000 / left 42.000000 / right 52.000000
05-31 08:35:52.717  2522  2522 I TrackingServiceClient: initialize
05-31 08:35:52.717  2522  2522 D TrackingServiceClient: onInitialize
05-31 08:35:52.718  2522  2522 I InputHooks: InitializeInputHooks
05-31 08:35:52.718  2522  2522 I InputHooks: InitializeInputHooks create
05-31 08:35:52.718  2522  2522 I InputHooks: InitializeInputHookscreated calling init
05-31 08:35:52.718  2522  2522 I InputHooks_Internal: SetReorientHMDOnControllerRecenter = false
05-31 08:35:52.718  2522  2522 I InputHooks_Internal: SetBlockRemoteButtonsWhenNotEmulatingHMT = true
05-31 08:35:52.718  2522  2522 I InputHooks_Internal: SetLatchBackButtonEntireFrame = true
05-31 08:35:52.718  2522  2522 I VrApi   : ----------- vrapi_Initialize [end] ----------
05-31 08:35:52.731  1152  1162 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.736  1152  1162 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.738  1152  1162 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.741  1152  1162 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.750  2522  2522 I VrApi   : ------------ vrapi_Shutdown [start] -----------
05-31 08:35:52.750  2522  2522 D TrackingServiceClient: onShutdown
05-31 08:35:52.750  2522  2522 I AudioCapture: AudioCapture::Shutdown
05-31 08:35:52.750  2522  2522 D SystemActivitiesReceiver: Unregistering Oculus System Activity receiver
05-31 08:35:52.751  2522  2522 I Breakpad: Breakpad Shutdown
05-31 08:35:52.751  2522  2522 I VrRuntimeClient: DisableGlFunctionHooks
05-31 08:35:52.751  2522  2522 I BufferObject: ovrBufferObjectFactory::Shutdown
05-31 08:35:52.751  2522  2522 I BufferObject: ovrBufferObjectFactory: closing fd=43
05-31 08:35:52.751  2522  2522 I RenderUtils: ovrRenderUtils::Shutdown - not initialized
05-31 08:35:52.751  2522  2522 D VrRuntimeClient: VrRuntimeClient: shutdown.
05-31 08:35:52.751  2522  2522 D VrRuntimeClient: VrRuntimeClient: Performing disconnect
05-31 08:35:52.751  2071  3431 D VrRuntimeService: VrRuntimeService: destroyClientInternal: com.android.packageinstaller, pid: 2522
05-31 08:35:52.751  2071  3431 I ClientMgr: ClientMgr::DestroyClient: 2522:com.android.packageinstaller
05-31 08:35:52.751  2071  3431 I ClientMgr: ClientMgr::DestroyClient: non-deferred ShutdownResources: 18446744073709551615 == 18446744073709551615
05-31 08:35:52.752  2071  3431 I BufferObject: ovrBufferObjectFactory::Shutdown
05-31 08:35:52.752  2071  3431 I BufferObject: ovrBufferObjectFactory: closing fd=44
05-31 08:35:52.752  2071  3431 I BufferObject: ovrBufferObjectFactory: unmapping Vulkan memory
05-31 08:35:52.752  2071  3431 I BufferObject: ovrBufferObjectFactory: destroying Vulkan buffer=547402949888
05-31 08:35:52.752  2071  3431 I BufferObject: ovrBufferObjectFactory: freeing Vulkan memory=547403525632
05-31 08:35:52.753  2522  2522 I VrRuntimeClient: nativeRuntimeServiceConnected: publicReadOnlyFd=-1, privateReadWriteFd=-1, bufferObjectFd=-1
05-31 08:35:52.755  4955  4983 I renderdoccmd: handle_cmd(10)
05-31 08:35:52.756  4955  4983 I renderdoccmd: handle_cmd(11)
05-31 08:35:52.757  2522  2522 I VrApi   : ------------ vrapi_Shutdown [end] -----------
05-31 08:35:52.758  1152  1523 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.759  4955  4983 I renderdoccmd: handle_cmd(13)
05-31 08:35:52.760  1152  1578 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.775  2071  2071 D VrRuntimeService: onFocusedWindowChanged: topWindowToken=android.os.BinderProxy@ea11a0b, topWindowUid=10084
05-31 08:35:52.775  2071  2071 W ClientMgr: ClientMgr::SetActiveCompositorClient: No active client
05-31 08:35:52.775  2071  2071 W VrRuntimeServer: VrRuntimeServer_SetClockLevels: Invalid Warp when trying to set -1 -1
05-31 08:35:52.777  4955  4983 I renderdoccmd: handle_cmd(1)
05-31 08:35:52.778  4955  4983 I renderdoccmd: APP_CMD_INIT_WINDOW
05-31 08:35:52.778  4955  4983 I renderdoccmd: spawning command thread
05-31 08:35:52.778  4955  4983 I renderdoccmd: Trying to splash
05-31 08:35:52.778  4955  4983 I renderdoccmd: Doing a splash
05-31 08:35:52.790  4955  4985 I renderdoccmd: Entering cmd thread
05-31 08:35:52.790  4955  4983 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.791  4955  4985 I renderdoc: @5cf12db800000002@ RDOC 004955: [08:35:52]    posix_network.cpp( 400) - Log     - Created and bind socket: 50
05-31 08:35:52.792  4955  4985 I renderdoc: @5cf12db800000004@ RDOC 004955: [08:35:52]    remote_server.cpp( 865) - Log     - No whitelist IP ranges configured - using default private IP ranges.
05-31 08:35:52.792  4955  4985 I renderdoc: @5cf12db800000006@ RDOC 004955: [08:35:52]    remote_server.cpp( 869) - Log     - Create a config file remoteserver.conf in ~/.renderdoc or %APPDATA%/renderdoc to narrow this down or accept connections from more ranges.
05-31 08:35:52.792  4955  4985 I renderdoc: @5cf12db800000008@ RDOC 004955: [08:35:52]    remote_server.cpp( 876) - Log     - Allowing connections from:
05-31 08:35:52.793  4955  4985 I renderdoc: @5cf12db80000000a@ RDOC 004955: [08:35:52]    remote_server.cpp( 885) - Log     - 10.0.0.0 / 255.0.0.0
05-31 08:35:52.793  4955  4985 I renderdoc: @5cf12db80000000c@ RDOC 004955: [08:35:52]    remote_server.cpp( 885) - Log     - 172.16.0.0 / 255.240.0.0
05-31 08:35:52.793  4955  4985 I renderdoc: @5cf12db80000000e@ RDOC 004955: [08:35:52]    remote_server.cpp( 885) - Log     - 192.168.0.0 / 255.255.0.0
05-31 08:35:52.793  4955  4985 I renderdoc: @5cf12db800000010@ RDOC 004955: [08:35:52]    remote_server.cpp( 889) - Log     - Allowing execution commands
05-31 08:35:52.793  4955  4985 I renderdoc: @5cf12db800000012@ RDOC 004955: [08:35:52]    remote_server.cpp( 893) - Log     - Replay host ready for requests...
05-31 08:35:52.796  4955  4983 I Adreno  : QUALCOMM build                   : b1f8328, I8a36212832
05-31 08:35:52.796  4955  4983 I Adreno  : Build Date                       : 03/05/19
05-31 08:35:52.796  4955  4983 I Adreno  : OpenGL ES Shader Compiler Version: EV031.26.02.00
05-31 08:35:52.796  4955  4983 I Adreno  : Local Branch                     :
05-31 08:35:52.796  4955  4983 I Adreno  : Remote Branch                    : refs/tags/AU_LINUX_ANDROID_LA.UM.5.7.R1.07.01.01.253.109
05-31 08:35:52.796  4955  4983 I Adreno  : Remote Branch                    : NONE
05-31 08:35:52.796  4955  4983 I Adreno  : Reconstruct Branch               : NOTHING
05-31 08:35:52.796  4955  4983 I Adreno  : Build Config                     : S L 3.8.9 AArch64
05-31 08:35:52.816  4955  4983 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.821  4955  4983 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.822  4955  4983 I Adreno  : PFP: 0x005ff112, ME: 0x005ff066
05-31 08:35:52.822  4955  4983 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.843  4955  4983 I renderdoccmd: Done splashing
05-31 08:35:52.843  4955  4983 I renderdoccmd: handle_cmd(12)
05-31 08:35:52.844  4955  4983 I renderdoccmd: handle_cmd(14)
05-31 08:35:52.844  4955  4983 I renderdoccmd: handle_cmd(6)
05-31 08:35:52.844  4955  4983 I renderdoccmd: doing misc splash
05-31 08:35:52.844  4955  4983 I renderdoccmd: Trying to splash
05-31 08:35:52.844  4955  4983 I renderdoccmd: Doing a splash
05-31 08:35:52.844  1152  1466 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.845  4955  4983 I Adreno  : QUALCOMM build                   : b1f8328, I8a36212832
05-31 08:35:52.845  4955  4983 I Adreno  : Build Date                       : 03/05/19
05-31 08:35:52.845  4955  4983 I Adreno  : OpenGL ES Shader Compiler Version: EV031.26.02.00
05-31 08:35:52.845  4955  4983 I Adreno  : Local Branch                     :
05-31 08:35:52.845  4955  4983 I Adreno  : Remote Branch                    : refs/tags/AU_LINUX_ANDROID_LA.UM.5.7.R1.07.01.01.253.109
05-31 08:35:52.845  4955  4983 I Adreno  : Remote Branch                    : NONE
05-31 08:35:52.845  4955  4983 I Adreno  : Reconstruct Branch               : NOTHING
05-31 08:35:52.845  4955  4983 I Adreno  : Build Config                     : S L 3.8.9 AArch64
05-31 08:35:52.846  1152  1549 E ActivityManager: Stack=ActivityStack{787b72e stackId=5, 1 tasks} isn't front stack but is at the top of the stack list
05-31 08:35:52.849  4955  4983 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.850  4955  4955 W IInputConnectionWrapper: reportFullscreenMode on inexistent InputConnection
05-31 08:35:52.851  4955  4983 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt
05-31 08:35:52.852  4955  4983 I Adreno  : PFP: 0x005ff112, ME: 0x005ff066
05-31 08:35:52.852  4955  4983 I Adreno  : Found an ESX Settings File: /data/misc/gpu/esx_config.txt

## Repro steps

Use Oculus Quest, in developer mode, try connecting via renderdoc.

## Environment

* RenderDoc 1.4
* Windows 10 Pro
* API: OpenGL

**Comment by baldurk** [comment]
I don't have any Oculus devices but you shouldn't need any special build as far as I know.

It looks like the remote server has started correctly and is listening, so if you can't connect then maybe there's an issue with the port forwarding. The output from adb logcat is awful to read (par for the course on android) - can you please upload the RenderDoc log instead as that also contains the PC-side logging?

**Comment by jeffkingdev** [comment]
AHA!  It was a port issue.  Thank you so much for the direction!  Once I opened those ports, it all started working.  Thanks again for your help, I'll close this.

**Comment by baldurk** [comment]
Can you clarify what you did exactly? What do you mean by "opened those ports"? All the port forwarding should be happening automatically via adb so if it failed then that's a bug to fix.

**Comment by jeffkingdev** [comment]
@baldurk Sorry, let me explain better.  My machine (still unbeknown to me how) gobbles up ports.  I noticed this when doing some software network development, that my TCP port I would use, wouldn't open because it was already being used -- not sure how.  Anyway, a remedy to that is just a simple reboot.  Once I tried that, the Quest connected right up.  I didn't even think of that until I saw your message about port forwarding which led me to believe it might be the same issue and it was.  I'm not sure if this helps anyone, but I thank you again for your help.  Next time I should try a reboot first 😄
